src/braze-upload/lambda/lambda.ts

```diff
@@ -84,33 +85,37 @@ export const processReferralCode = async (referralCode: string): Promise<void> =
             referralCode: referralCode,
             campaignId: referralData.campaign_id,
         },
-        pool
+        dbConnectionPool
     );
     if (writeResult.rows.length <= 0) {
         return Promise.reject(`Failed to write successful referral for code ${referralCode}`);
     }
 
     // Fetch the distinct set of campaignIds for this braze user
-    const campaignIdsResult: QueryResult = await fetchCampaignIds(referralData.braze_uuid, pool);
+    const campaignIdsResult: QueryResult = await fetchCampaignIds(referralData.braze_uuid, dbConnectionPool);
     if (campaignIdsResult.rows.length <= 0) {
         return Promise.reject(`No campaignIds found for brazeUuid ${referralData.braze_uuid}`);
     }
 
     const campaignIds = campaignIdsResult.rows.map(row => row.campaign_id);
 
-    return sendCampaignIdsToBraze(campaignIds, referralData.braze_uuid, config.brazeKey);
+    return sendCampaignIdsToBraze(campaignIds, referralData.braze_uuid, brazeKey);
 };
 
-export async function handler(event: Event, context: any): Promise<any> {
-
-    const maybeReferralCodes: (string | null)[] = await Promise.all(
-        event.Records.map(record => getReferralCodeFromThriftBytes(record.kinesis.data))
-    );
-
-    const resultPromises = maybeReferralCodes
-        .filter(maybeReferralCode => !!maybeReferralCode)
-        .map(c => c as string)  // typescript doesn't know that the above line filters to strings only
-        .map(processReferralCode);
-
-    return Promise.all(resultPromises);
+export function handler(event: Event, context: any, callback: (err: Error | null, result?: number) => void) {
+    // setTimeout is necessary because of a bug in the node lambda runtime which breaks requests to ssm
+    setTimeout(async () => {
+        const maybeReferralCodes: (string | null)[] = await Promise.all(
+            event.Records.map(record => getReferralCodeFromThriftBytes(record.kinesis.data))
+        );
+
+        const resultPromises = maybeReferralCodes
+            .filter(maybeReferralCode => !!maybeReferralCode)
+            .map(c => c as string)  // typescript doesn't know that the above line filters to strings only
+            .map(processReferralCode);
+
+        Promise.all(resultPromises)
```

Comment:
do you need `return await` here?

Reply:
Here we are now 'side-effecting' rather than returning a value.
So we use the given callback to complete the lambda run (whereas before we returned a promise)

Reply:
Then does the function called by the setTimeout still need to be async?

Reply:
yes, but only because I `await` on line 108. Otherwise it would not need to be async. But I think in this case the use of `await` makes it cleaner

Reply:
Ah, that's what I suspected. It just jarred with me to see an async anonymous function declared within a setTimeout - mixing world orders!

Reply:
Yes it is odd, the `setTimeout` is unfortunate! Hopefully AWS will fix it soon...